src/xrpld/app/misc/WamrVM.cpp

```diff
@@ -640,6 +640,62 @@ WamrEngine::call(
     return call<NR>(func, in, p.data(), p.size(), std::forward<Types>(args)...);
 }
 
+Expected<int32_t, NotTEC>
+WamrEngine::preflight(
+    wbytes const& wasmCode,
+    std::string_view funcName,
+    std::vector<WasmImportFunc> const& imports,
+    std::vector<WasmParam> const& params)
+{
+    try
+    {
+        return preflightHlp(wasmCode, funcName, imports, params);
+    }
+    catch (std::exception const& e)
+    {
+        std::cerr << "WAMR: preflight exception: " << e.what() << std::endl;
+    }
+    catch (...)
+    {
+        return temUNKNOWN;
+    }
+    return Unexpected<NotTEC>(temBAD_WASM);
+}
+
+Expected<int32_t, NotTEC>
+WamrEngine::preflightHlp(
```

Comment:
This function is exact as runHlp, why you duplicate it?

Reply:
It was different for a bit when writing this PR, and I assume it'll be different in the future (e.g. with the suggestion you gave me on that API function)

Reply:
Also, different error codes